lib/galaxy/webapps/galaxy/api/history_contents.py

```diff
@@ -486,21 +486,17 @@ def archive(self, trans, history_id, filename='', format='zip', dry_run=True, **
         return self.service.archive(trans, history_id, filter_parameters, filename, dry_run)
 
     @expose_api_raw_anonymous
-    def contents_near(self, trans, history_id, hid, limit, **kwd):
+    def contents_near(self, trans, history_id, direction, hid, limit, **kwd):
         """
-        This endpoint provides random access to a large history without having
-        to know exactly how many pages are in the final query. Pick a target HID
-        and filters, and the endpoint will get LIMIT counts above and below that
-        target regardless of how many gaps may exist in the HID due to
-        filtering.
+        Return {limit} history items "near" the {hid}. The {direction} determines what items
+        are selected:
+        - before: select items with hid < {hid}
+        - after:  select items with hid > {hid}
+        - near:   select items "around" {hid}, so that |before| <= limit // 2, |after| <= limit // 2 + 1
 
-        It does 2 queries, one up and one down from the target hid with a
-        result size of limit. Additional counts for total matches of both seeks
-        provided in the http headers.
+        Additional counts provided in the HTTP headers.
 
-        I've also abandoned the wierd q/qv syntax.
```

Comment:
Now that we are modifying this endpoint... I know Mason may have his reasons to abandon the q/qv syntax, I also found it strange at the beginning, but the thing is, the q/qv syntax can be modeled and documented in FastAPI, while for the new one there is no way to document every possible combination of parameters and also requires to do some [little hacks accessing the request query parameters directly](https://github.com/galaxyproject/galaxy/pull/12578/commits/e7d355bcfb80feecbe5700e85ecc0ee0805f8cb3#diff-d7789a8b5b8606bd040d59409203acc178c14fe4ec4d6b38e8955d87dc4ecea1R290-R301) while is changing how the rest of the endpoints behave.

But definitely, not something that should be addressed in this PR, I was commenting just to get a general opinion now that we have stumbled upon it again, and maybe I can address it in a follow-up PR if people agree.

Reply:
Marking this as resolved, as the discussion moved to matrix.